### ISSUE TITLE
Fails after about hour of working

### ISSUE THREAD

**Issue by unclechu** [daa84/neovim-gtk#24]
It freezes and after about 10-20 seconds it fails.

Here is the log:
```
$ nvim-gtk 
Unknown error type: Wait timeout
Unknown error type: Wait timeout
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: "SendError(..)"', /checkout/src/libcore/result.rs:906:4
note: Run with `RUST_BACKTRACE=1` for a backtrace.
thread '<unnamed>' panicked at 'Can't join dispatch thread: Any', /checkout/src/libcore/result.rs:906:4
Unknown error type: Wait timeout
Unknown error type: Wait timeout
nUnknown error type: Wait timeout
Unknown error type: Wait timeout
thread 'main' panicked at 'Error run input command to nvim: GenericError("Wait timeout")', /checkout/src/libcore/result.rs:906:4
```
Neovim `:version`:
```viml
NVIM v0.2.2
Build type: Debug
LuaJIT 2.0.5
Compilation: /usr/bin/cc -Wconversion -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -I/home/unclechu/.compile_n_build/neovim/neovim-v0.2.2-from-git/build/config -I/home/unclechu/.compile_n_build/neovim/neovim-v0.2.2-from-git/src -I/home/unclechu/.compile_n_build/neovim/neovim-v0.2.2-from-git/.deps/usr/include -I/home/unclechu/.compile_n_build/neovim/neovim-v0.2.2-from-git/.deps/usr/include -I/home/unclechu/.compile_n_build/neovim/neovim-v0.2.2-from-git/.deps/usr/include -I/home/unclechu/.compile_n_build/neovim/neovim-v0.2.2-from-git/.deps/usr/include -I/home/unclechu/.compile_n_build/neovim/neovim-v0.2.2-from-git/.deps/usr/include -I/home/unclechu/.compile_n_build/neovim/neovim-v0.2.2-from-git/.deps/usr/include -I/usr/include -I/home/unclechu/.compile_n_build/neovim/neovim-v0.2.2-from-git/build/src/nvim/auto -I/home/unclechu/.compile_n_build/neovim/neovim-v0.2.2-from-git/build/include
Compiled by unclechu@fedorafarm25

Features: +acl +iconv +jemalloc +tui 
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/home/unclechu/neovim-v0.2.2-from-git/share/nvim"

Run :checkhealth for more info
```
My neovim config: https://github.com/unclechu/neovimrc/

**Comment by daa84** [comment]
Hello!
Wait timeout error occur when neovim process does not respond in 5 seconds. This can be:
* some bug in neovim
* just long response time because of something, maybe neovim was too busy
* this can be some unsupported neovim behavior (for example neovim process wait for something that neovim-gtk don't know)
One solution is maybe increase 5 seconds to some bigger value.

**Comment by unclechu** [comment]
@daa84 where this timeout is set so I could change it?

**Comment by wezm** [comment]
I just encountered the same crash but the cause may be different. I'm not actually sure what triggered it as I wasn't using Neovim at the time.

```
Nov 24 13:45:11 wes-thinkpad org.daa.NeovimGtk.desktop[21104]: Unknown error type: Wait timeout
Nov 24 13:45:16 wes-thinkpad org.daa.NeovimGtk.desktop[21104]: Unknown error type: Wait timeout
Nov 24 13:45:21 wes-thinkpad org.daa.NeovimGtk.desktop[21104]: Unknown error type: Wait timeout
Nov 24 13:45:26 wes-thinkpad org.daa.NeovimGtk.desktop[21104]: Unknown error type: Wait timeout
Nov 24 13:45:31 wes-thinkpad org.daa.NeovimGtk.desktop[21104]: Unknown error type: Wait timeout
Nov 24 13:45:36 wes-thinkpad org.daa.NeovimGtk.desktop[21104]: Unknown error type: Wait timeout
Nov 24 13:45:36 wes-thinkpad org.daa.NeovimGtk.desktop[21104]: thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: "SendError(..)"', /checkout/src/libcore/result.rs:860:4
Nov 24 13:45:36 wes-thinkpad org.daa.NeovimGtk.desktop[21104]: note: Run with `RUST_BACKTRACE=1` for a backtrace.
Nov 24 13:45:36 wes-thinkpad org.daa.NeovimGtk.desktop[21104]: thread '<unnamed>' panicked at 'Can't join dispatch thread: Any', /checkout/src/libcore/result.rs:860:4
Nov 24 13:45:41 wes-thinkpad org.daa.NeovimGtk.desktop[21104]: Unknown error type: Wait timeout
Nov 24 13:45:46 wes-thinkpad org.daa.NeovimGtk.desktop[21104]: Unknown error type: Wait timeout
Nov 24 13:45:51 wes-thinkpad org.daa.NeovimGtk.desktop[21104]: Unknown error type: Wait timeout
Nov 24 13:45:56 wes-thinkpad org.daa.NeovimGtk.desktop[21104]: Unknown error type: Wait timeout
Nov 24 13:46:01 wes-thinkpad org.daa.NeovimGtk.desktop[21104]: Unknown error type: Wait timeout
Nov 24 13:46:06 wes-thinkpad org.daa.NeovimGtk.desktop[21104]: Unknown error type: Wait timeout
Nov 24 13:46:11 wes-thinkpad org.daa.NeovimGtk.desktop[21104]: thread 'main' panicked at 'Can't send mouse input event: GenericError("Wait timeout")', /checkout/src/libcore/result.rs:860:4
```

I got two "NeovimGtk isn't responing" dialogues from GNOME before it crashed. They give the choice of Force Quit or Wait, I chose Wait in each case.

**Comment by daa84** [comment]
I found same issue in neovim-qt: https://github.com/equalsraf/neovim-qt/issues/319
So I think it is need to introduce new --timeout command line argument like neovim-qt do and increase default timeout, maybe this helps. 
@unclechu currently it is not possible to change timeout value

**Comment by daa84** [comment]
Fixed memory leak #28 - this can produce slowdown and timeout error at the end.
Fixed problem with timeout error when neovim wait for user input and focus event executed.
Also increase default timeout to 10 sec and add option to set timeout from command line (https://github.com/daa84/neovim-gtk/wiki/Configuration#command-line).
Hope this helps 😄
